Assets/Scripts/Character.cs

```diff
@@ -0,0 +1,28 @@
+using UnityEngine;
+
+public class Character : MonoBehaviour {
+    public float movementSpeed = 10f;
+    public float rotationSpeed = 720f;
+
+    private void Update() {
+        var x = Input.GetAxis("Horizontal");
+        var z = Input.GetAxis("Vertical");
+        var movementDirection = new Vector3(x, 0, z);
+        movementDirection.Normalize();
+
+        var currentVelocity = movementDirection * movementSpeed;
+        var currentPosition = transform.position;
+        var nextPosition = Vector3.SmoothDamp(
+            currentPosition,
+            currentPosition + movementDirection * movementSpeed * Time.deltaTime,
+            ref currentVelocity,
+            Time.deltaTime
+        );
+        transform.position = nextPosition;
+
+        if (movementDirection == Vector3.zero) return;
```

Comment:
direct comparisons with zero are not correct (because of floating-point specifics) 
you should check whether a magnitude is below a certain threshold